### ISSUE TITLE
LWLibavSource crashes with specific file

### ISSUE THREAD

**Issue by ChrisK2** [VFR-maniac/L-SMASH-Works#42]
When trying to decode [sample.m2ts](http://sva.wakku.to/~chris/perm_stuff/crashsamples/sample.m2ts) with latest LWLibavSource, it immediately crashes python.
mpv plays the file fine and without decoder errors. FFMS2 also decodes it but discards the first frame (seems to be another bug).

[OS X crash report](http://sva.wakku.to/~chris/perm_stuff/crashsamples/crashreport.txt)
[lwi file](http://sva.wakku.to/~chris/perm_stuff/crashsamples/sample.m2ts.lwi.zip)

Script:
```
import sys
from subprocess import Popen, PIPE

import vapoursynth as vs
core = vs.get_core()

file = str("sample.m2ts")

video = core.lsmas.LWLibavSource(source=file)
video = core.std.Trim(clip=video, first=0, last=99)

#video.set_output(0)

cmd = ['x264', '-', '--frames', '100', '--input-csp', 'i420', '--demuxer', 'raw', '--input-depth', '8', '--input-res', '1920x1080', '--fps', '24000/1001', '-o', 'testencode.mp4', '--colorprim', 'bt709', '--colormatrix', 'bt709']

p = Popen(cmd, stdin=PIPE)
video.output(p.stdin)
p.communicate()
```

Using vspipe also leads to a crash.

**Comment by ChrisK2** [comment]
Also happens on Linux (ubuntu). Too bad nobody cares.

**Comment by MaverickTse** [comment]
I haven't test on Linux yet, but Windows 32bit build works with the VapourSynth R28 pre-built binaries with Python 3.5 (32bit)
![vps_32working_lsw](https://cloud.githubusercontent.com/assets/10569988/10532225/4fa69a94-73ee-11e5-839d-882a3b8c90b0.PNG)

**Comment by ChrisK2** [comment]
Thanks for testing. I used self-compiled 64 bit Binaries.

**Comment by sl1pkn07** [comment]
i get this on my archlinux 64bits

    vspipe: src/core/vslog.cpp:73: void vsLog(const char*, long int, VSMessageType, const char*, ...): Assertion `false' failed.

VS R30 (from Git) and lsmashsource r868

with VSEdit can decode the firts 16 frames, in the frame 17,18 and 19 crash vsedit. the rest also can decode ok (from 20 to 99)

**Comment by dubhater** [comment]
@sl1pkn07: The message printed just before that is what matters.

**Comment by sl1pkn07** [comment]
└───╼  vspipe -y lsmash-test.vpy - | mpv -
    Playing: -
    [file] Reading from stdin...
     (+) Video --vid=1 (rawvideo)
    No frame returned at the end of processing by LWLibavSource
    vspipe: src/core/vslog.cpp:73: void vsLog(const char*, long int, VSMessageType, const char*, ...): Assertion `false' failed.
    Using software decoding.
    VO: [opengl-hq] 1920x1080 yuv420p
    V: 00:00:00 Cache:  0s+0KB
    
    
    Exiting... (End of file)

**Comment by myrsloik** [comment]
Try again with the latest lsmashsource. It should at least return a proper error now instead of crashing.

**Comment by sl1pkn07** [comment]
Error: Failed to retrieve frame 17 with error: lsmas: failed to output a video frame.

**Comment by myrsloik** [comment]
And the other obvious question that will probably help me and/or the author. Does it work in avisynth?

**Comment by sl1pkn07** [comment]
can confirm works in avsplus r1825 MT (32bits)

but the frames 17, 18 and 19 have weird color and i think, corruptions

![frame 16](http://wstaw.org/m/2016/01/31/Screenshot_20160131_235318.png)
![frame 17](http://wstaw.org/m/2016/01/31/Screenshot_20160131_235325.png)
![frame 18](http://wstaw.org/m/2016/01/31/Screenshot_20160131_235331.png)
![frame 19](http://wstaw.org/m/2016/01/31/Screenshot_20160131_235338.png)
![frame 20](http://wstaw.org/m/2016/01/31/Screenshot_20160131_235343.png)

i think the main problem is in the source instead of the lsmashsource

**Comment by myrsloik** [comment]
Yes, it is. In avisynth it simply returns an uninitialized frame for many of the possible errors. So you get random memory which is usually a recycled previous frame. Definitely not what's expected either. If anything returning the most recent ok frame makes more sense if you want to do error concealing.

**Comment by sl1pkn07** [comment]
ffms2 on vs don't have this problem, all frames docode ok

**Comment by sl1pkn07** [comment]
Nope, with ffms2 eats that bad frames

lsmashsource (VS)
![frame22](http://wstaw.org/m/2016/02/12/Screenshot_20160212_005432.png)

and same frame whit ffms2 (VS)
![frame22](http://wstaw.org/m/2016/02/12/Screenshot_20160212_005451.png)

same behavior in (avs)
![frame22](http://wstaw.org/m/2016/02/12/Screenshot_20160212_005841.png)
![frame22](http://wstaw.org/m/2016/02/12/Screenshot_20160212_005829.png)

all after removing the index files

**Comment by divxmaster** [comment]
I have found a workaround for this issue.
My platform is 64 bit windows/64 bit VS, non Linux, but I get the same  lsmas: failed to output a video frame error on most m2ts files that I open with vs ffms2 - at the same frame, 17-19.  It only occurs on m2ts files that have been remuxed with tsmuxer or clownbd. If I open the original 00800.m2ts or whatever, the error does not occur.  The workaround is to use BDREBUILDER.  If I remux (set to 50gb output), which will direct stream copy with this option, the error does not occur. Although bdrb uses tsmuxer, somehow it is using it differently or specifying different parameters.  This doesn't help if you have already created the m2ts remuxes of course.  I must also get around to trying a pure 32bit VS environment.

Cheers,
Divxmaster

**Comment by divxmaster** [comment]
Also found a way to fix m2ts files that you have already processed with tsmuxer or clownbd.  In bdrebuilder, use the import -> video files option. This takes a while, but when it finally processes/loads it will have created a new 00000.m2ts file in the working directories. There is no need to tell bdrebuilder to 'save', you can just grab the temporary file.  This file works fine with lsmas.
